src/libraries/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -518,6 +518,43 @@ public void UnixDomainSocketEndPoint_RelativePathDeletesFile()
             }).Dispose();
         }
 
+        [ConditionalFact(nameof(PlatformSupportsUnixDomainSockets))]
+        public void AbstractPathEquality()
+        {
+            string abstractPath = '\0' + Guid.NewGuid().ToString();
+            UnixDomainSocketEndPoint endPoint1 = new(abstractPath);
+            UnixDomainSocketEndPoint endPoint2 = new(abstractPath);
+            UnixDomainSocketEndPoint endPoint3 = new('\0' + Guid.NewGuid().ToString());
+
+            Assert.Equal(endPoint1, endPoint2);
+            Assert.Equal(endPoint1.GetHashCode(), endPoint2.GetHashCode());
+
+            Assert.NotEqual(endPoint1, endPoint3);
+            Assert.NotEqual(endPoint2, endPoint3);
+            Assert.NotEqual(endPoint1.GetHashCode(), endPoint3.GetHashCode());
+            Assert.NotEqual(endPoint2.GetHashCode(), endPoint3.GetHashCode());
```

Comment:
These hashcode NotEqual asserts could sporadically fail.  It should be exceedingly rare, but better to remove them.  We could instead assert that the endpoint hashcode equals the path hashcode, and leave the correctness of the path hashcode up to string's implementation.

Same for the test below.

Reply:
Because of hash code collision.. hmm, makes sense. I would prefer just to remove `NotEqual` assertions.